sipn_reanalysis_ingest/luigitasks/convert.py

```diff
@@ -38,67 +44,112 @@ class Grib2ToNcDaily(luigi.Task):
 
     date = luigi.DateParameter()
 
+    @property
+    def uses_5day_tars(self):
+        if self.date < CFSR_DAILY_TAR_BEFORE:
+            return True
+        return False
+
     def requires(self):
-        five_day_window = Cfsr5ishDayWindow.from_date_in_window(self.date)
-
-        req = {
-            CfsrGranuleProductType.ANALYSIS: UntarCfsr5DayFile(
-                window_start=five_day_window.start,
-                window_end=five_day_window.end,
-                product_type=CfsrGranuleProductType.ANALYSIS,
-            ),
-            CfsrGranuleProductType.FORECAST: UntarCfsr5DayFile(
-                window_start=five_day_window.start,
-                window_end=five_day_window.end,
-                product_type=CfsrGranuleProductType.FORECAST,
-            ),
-        }
-
-        # If the first day of analysis window, we also need the forecast files for the
-        # previous window!
-        if self.date == five_day_window.start:
-            forecast_5day_window = Cfsr5ishDayWindow.from_date_in_window(
-                self.date - dt.timedelta(days=1)
-            )
-            req[CfsrGranuleProductType.FORECAST] = [
-                req[CfsrGranuleProductType.FORECAST],
-                UntarCfsr5DayFile(
-                    window_start=forecast_5day_window.start,
-                    window_end=forecast_5day_window.end,
+        if self.uses_5day_tars:
+            five_day_window = Cfsr5ishDayWindow.from_date_in_window(self.date)
+
+            req = {
+                CfsrGranuleProductType.ANALYSIS: UntarCfsr5DayFile(
+                    window_start=five_day_window.start,
+                    window_end=five_day_window.end,
+                    product_type=CfsrGranuleProductType.ANALYSIS,
+                ),
+                CfsrGranuleProductType.FORECAST: UntarCfsr5DayFile(
+                    window_start=five_day_window.start,
+                    window_end=five_day_window.end,
                     product_type=CfsrGranuleProductType.FORECAST,
                 ),
-            ]
+            }
 
-        return req
+            # If the first day of analysis window, we also need the forecast files for the
+            # previous window!
+            if self.date == five_day_window.start:
+                forecast_5day_window = Cfsr5ishDayWindow.from_date_in_window(
+                    self.date - dt.timedelta(days=1)
+                )
+                req[CfsrGranuleProductType.FORECAST] = [
+                    req[CfsrGranuleProductType.FORECAST],
+                    UntarCfsr5DayFile(
+                        window_start=forecast_5day_window.start,
+                        window_end=forecast_5day_window.end,
+                        product_type=CfsrGranuleProductType.FORECAST,
+                    ),
+                ]
+
+            return req
+        else:
```

Comment:
We'll need an `elif` here to handle the special April 1st case. In that case, we need the current day's daily tar to get 4 analysis gribs and 3 forecast gribs, and the previous day's _forecast_ 5-day tar to get 1 forecast grib?